### ISSUE TITLE
DocumentService Lifecycles/Middlewares cannot be registered

### ISSUE THREAD

**Issue by derrickmehaffy** [strapi/strapi#19998]
### Required System information

- Node.js version: v20.11.1
- Yarn version: 1.22.22
- Strapi version: 5.0.0-beta.1
- Database: SQLite
- Operating system: Linux Mint 21.3
- Is your project Javascript or Typescript: Javascript

### Describe the bug

When trying to register a new documentService lifecycle around any of the functions an error is thrown that breaks the content-manager: `[2024-04-02 11:53:59.023] error: middlewares[(index++)] is not a function`

### Steps to reproduce the behavior

1. Create a new Strapi 5 beta project
2. Try to register a new documentService Lifecycle in the `./src/index.js` (See code example below)
3. Start the application and try and go to the content-manager

### Expected behavior

Should register the lifecycle

### Screenshots

![image](https://github.com/strapi/strapi/assets/8593673/c6958b47-bd67-4e21-ba12-8b09525d0b37)

```
[2024-04-02 12:00:42.831] error: middlewares[(index++)] is not a function
TypeError: middlewares[(index++)] is not a function
    at next (/mnt/storage/Development/testLifecyclesV5/node_modules/@strapi/core/dist/services/document-service/middlewares/middleware-manager.js:14:38)
    at Array.databaseErrorsMiddleware (/mnt/storage/Development/testLifecyclesV5/node_modules/@strapi/core/dist/services/document-service/middlewares/errors.js:13:18)
    at next (/mnt/storage/Development/testLifecyclesV5/node_modules/@strapi/core/dist/services/document-service/middlewares/middleware-manager.js:14:38)
    at Object.run (/mnt/storage/Development/testLifecyclesV5/node_modules/@strapi/core/dist/services/document-service/middlewares/middleware-manager.js:18:14)
    at Object.newMethod [as findMany] (/mnt/storage/Development/testLifecyclesV5/node_modules/@strapi/core/dist/services/document-service/middlewares/middleware-manager.js:32:30)
    at Object.findPage (/mnt/storage/Development/testLifecyclesV5/node_modules/@strapi/plugin-content-manager/server/src/services/document-manager.ts:86:31)
    at Object.find (/mnt/storage/Development/testLifecyclesV5/node_modules/@strapi/plugin-content-manager/server/src/controllers/collection-types.ts:135:70)
    at runNextTicks (node:internal/process/task_queues:60:5)
    at process.processImmediate (node:internal/timers:449:9)
    at process.topLevelDomainCallback (node:domain:160:15)
[2024-04-02 12:00:42.832] http: GET /content-manager/collection-types/api::test.test?page=1&pageSize=10&sort=test:ASC (21 ms)
```

### Code snippets

```js
// path: ./src/index.js

"use strict";

module.exports = {
  register(/*{ strapi }*/) {},

  bootstrap(/*{ strapi }*/) {
    strapi.documents.use("findMany", async (ctx, next) => {
      console.log("test");
    });
  },
};
```

### Additional context

Related to: https://github.com/strapi/strapi/issues/19982

**Comment by alexandrebodin** [comment]
Hi, this is an invlid issue, 

The document service middlewares have to be used as follows:

```js
strapi.documents.use((ctx, next) => {
	
	// ctx.action gives the names of the action
	// ctx.uid gives the name of the content-type uid
	// ctx.args is the array of arguments received by the called action
	
	return next(ctx);
})
```

**Comment by laurenskling** [comment]
> Hi, this is an invlid issue,
> 
> The document service middlewares have to be used as follows:
> 
> ```js
> strapi.documents.use((ctx, next) => {
> 	
> 	// ctx.action gives the names of the action
> 	// ctx.uid gives the name of the content-type uid
> 	// ctx.args is the array of arguments received by the called action
> 	
> 	return next();
> })
> ```

testing with this and it's actually `ctx.contentType` with the entire contentType object, not `ctx.uid`

**Comment by laurenskling** [comment]
Wouldn't it make sense to do this?:
```
strapi.documents('api::my-content-type').use((ctx, next) => {
  return next();
})
```

**Comment by derrickmehaffy** [comment]
I rewrote one of my old examples that switched from model lifecycles to the docsrv middlewares: https://github.com/strapi-support-demo-apps/strapi-example-v5-injectUUID

There are a few other issues/improvements I noticed but so far it's not too terrible to work with.

I'm with you that it would be nice to only specifically inject a middleware for a specific uid or have a way to provide a list of uids to pre-filter (the dynamic model lifecycles using subscribe has that).

The other thing I'm not a fan of is the structure of the `args: []` since it's an array with either just an object or a string (documentId) + the object. Makes it tricky to work with easily.
